core/src/main/java/io/questdb/griffin/engine/functions/catalogue/ShowTablesFunctionFactory.java

```diff
@@ -253,38 +249,15 @@ private boolean open(TableToken tableToken) {
                         }
                     }
 
-                    if (getMetadata() == METADATA) {
-                        TableToken lastTableTokenVersion = executionContext.getCairoEngine().getTableTokenIfExists(tableToken.getTableName());
-                        if (lastTableTokenVersion != null) {
-                            tableToken = lastTableTokenVersion;
-                            try (TableMetadata tableMetadata = executionContext.getMetadataForRead(tableToken)) {
-                                isSoftLink = tableMetadata.isSoftLink();
-                                maxUncommittedRows = tableMetadata.getMaxUncommittedRows();
-                                o3MaxLag = tableMetadata.getO3MaxLag();
-                                partitionBy = tableMetadata.getPartitionBy();
-                                int timestampIndex = tableMetadata.getTimestampIndex();
-                                timestampColumnName = timestampIndex > -1 ? tableMetadata.getColumnName(timestampIndex) : null;
-                                isDedup = timestampIndex >= 0 && tableToken.isWal() && tableMetadata.isDedupKey(timestampIndex);
-                            } catch (CairoException e) {
-                                // perhaps this folder is not a table remove it from the result set
-                                LOG.info()
-                                        .$("cannot query table metadata [table=").$(tableToken)
-                                        .$(", error=").$(e.getFlyweightMessage())
-                                        .$(", errno=").$(e.getErrno())
-                                        .I$();
-                                return false;
-                            } catch (TableReferenceOutOfDateException e) {
-                                LOG.info()
-                                        .$("cannot query table metadata, can be concurrent table drop [table=").$(tableToken)
-                                        .$(", error=").$(e.getFlyweightMessage())
-                                        .I$();
-                                return false;
-                            }
-                        } else {
-                            // Table is dropped.
-                            return false;
-                        }
+                    table = CairoMetadata.INSTANCE.getTableQuiet(tableToken);
```

Comment:
this concoction has to be single reliable call:

- table is Not Null if it isn't deleted on-the-fly
- table is Null if it is deleted

the callee need not care about hydrating the cache